[Community] 커뮤니티 관련 컴포넌트 초안

kelvin-chihyun opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?

> 어떤 기능을 구현한건지, 이슈 대응이라면 어떤 이슈인지 PR이 열리게 된 계기와 목적을 Reviewer 들이 쉽게 이해할 수 있도록 적어 주세요
> 일감 백로그 링크나 다이어그램, 피그마를 첨부해도 좋아요

- 커뮤니티 관련 페이지(Community / CommunityDetail / CommunityForm / Admin)의 기능 및 레이아웃을 구현하였습니다.
- 파이어베이스의 Fireauth, FireStore, Stroage 서비스를 활용해 더미데이터를 작성하여 CRUD 기능을 적용하였습니다.

## 1. Community(Read)
![커뮤니티1](https://github.com/ReOpenSG/ReOpenSG/assets/101504272/bf9a8944-f9aa-400b-aaac-2e53751e17a4)

## 2. CommunityDetail
![커뮤니티상세1](https://github.com/ReOpenSG/ReOpenSG/assets/101504272/d633f56c-58d3-4170-b73e-d99f25e0353f)

## 3-1. CommunityForm(Create)
![커뮤니티 글쓰기](https://github.com/ReOpenSG/ReOpenSG/assets/101504272/ae9f4bff-2269-4175-9f0b-94d887eebb23)

## 3-2. CommunityForm(Update)
![커뮤니티 수정하기](https://github.com/ReOpenSG/ReOpenSG/assets/101504272/31a95659-04d9-481e-8d57-29771ecf7794)

## 3-3. CommunityForm(Delete)
![커뮤니티 삭제하기](https://github.com/ReOpenSG/ReOpenSG/assets/101504272/a58f8375-78c8-48fa-97e2-788c008c400e)

## 4. Admin 
![어드민1](https://github.com/ReOpenSG/ReOpenSG/assets/101504272/3627aa21-09b3-4027-b0b5-d729068c8d1b)

## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?

> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요

- 반응형을 적용하였습니다.
- CSS 모듈화하였습니다. (추후 페이지 컴포넌트도 적용 가능성 O)
- 컴포넌트를 분리하였습니다. (추후 추가 컴포넌트 분리 예정입니다.)

## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?

> 없으면 "없음" 이라고 기재해 주세요

-

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요

> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요

-

## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?

> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요

-

### 📌 PR 진행 시 이러한 점들을 참고해 주세요

- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 3일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
  - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
  - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
  - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

#

---

## 📝 Assignee를 위한 CheckList

- [ ] To-Do Item

kelvin-chihyun committed:
✨ [Feat] : 커뮤니티 페이지 일부 구현

커뮤니티 페이지 중 게시글 리스트 페이지를 구현하였습니다.
더미 데이터로 페이지네이션 기능도 구현하였으며, 데이터는 추후에
API로 받아오도록 수정할 예정입니다.

Related to #1 #2

kelvin-chihyun committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

kelvin-chihyun committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

kelvin-chihyun committed:
✨ [Feat] : 커뮤니티 페이지 일부 구현

커뮤니티 페이지 중 게시글 상세 페이지를 구현하였습니다.
게시글 리스트 페이지로부터 데이터를 state로 넘겨받아 렌더링하며,
데이터는 추후에 API로 받아오도록 수정할 예정입니다.

Related to #1, #3

kelvin-chihyun committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

kelvin-chihyun committed:
🐛 [Fix] : 이전, 다음페이지 이동 오류 해결

params.id에 따라 해당하는 게시글이 불러와지지 않는 오류를
UseEffect와 UseState를 활용하여 해결했습니다.

Related to #1 #3

kelvin-chihyun committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

kelvin-chihyun committed:
✨ [Feat] : Admin 기능 추가

1. admin 페이지를 만들었습니다.
2. 게시판(커뮤니티)에 글을 CRUD하기 위하여,
firebase의 auth 계정 생성후 설정하였습니다.
3. admin 계정 로그인 여부와 인증 유지를 전역에서 상태관리 하기 위하여,
recoil로 auth를 추적 관리하게 설정하였습니다.

Related to #1 #4

kelvin-chihyun committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

kelvin-chihyun committed:
✨ [Feat] : 커뮤니티 게시판 Create, Read 기능 추가

1. Firebase로부터 게시글들을 읽어오는 Read 기능을 추가하였습니다.
2. 로그인 후 글쓰기를 통해, Firebase에 Create하고, List가 추적할 수 있게끔 하였습니다.

Related to #1 #2 #4

kelvin-chihyun committed:
✨ [Feat] : 게시판 Update, Delete 기능 추가

1. 게시판 Edit 컴포넌트를 작성하였습니다.
2. 게시판 Detail 컴포넌트에서 수정, 삭제 버튼을 추가하였습니다.
3. 수정, 삭제 로직을 만들어, Firebase에 반영되게 하였습니다.

Related to #1 #2 #3 #4

kelvin-chihyun committed:
🐛 [Fix] : 게시글 CRUD 간 오류 수정

1. 새 글 작성시 이전, 다음 글이 정렬이 안되는 오류를 수정하였습니다.
2. 게시글 수정시 내용 업데이트가 안되는 오류를 수정하였습니다.

Related to #1 #4

kelvin-chihyun committed:
♻️ [Refactor] : Community 부분 컴포넌트 분리

Community(리스트) 컴포넌트, CommunityDetail(상세) 컴포넌트, CommunityForm(추가/수정) 컴포넌트를
section 단위로 구분하여 컴포넌트 분리하였습니다.
이후, CSS 모듈화, 반응형 CSS를 추가하여 수정할 예정입니다.

Related to #1 #2 #3

kelvin-chihyun committed:
💬 [Comment] : 주석 삭제

InputSection, ListButtons 컴포넌트에서 주석을 삭제하였습니다.

Related to #1

kelvin-chihyun committed:
🌈 [Design] : 반응형 추가

Community, CommunityDetail, CommunnityForm 컴포넌트에
반응형을 추가하였습니다.
디테일한 부분에서 향후 수정될 소요가 있습니다.

Related to #40

kelvin-chihyun committed:
🌈 [Design] : CSS 모듈화 일부

Community, CommunityForm 컴포넌트의 하위 컴포넌트를
CSS 모듈화
하였습니다.

Related to #40

kelvin-chihyun committed:
🌈 [Design] : CSS 모듈화

Admin 페이지 컴포넌트를 CSS 모듈화 하였습니다.

Related to #40

kelvin-chihyun committed:
🐛 [Fix] : prop-types 변경

Notice, CommmunityForm prop을 잘못 지정하여,
defaultProp 등을 설정하여 조치하였습니다.

Related to #1

kelvin-chihyun committed:
🌈 [Design] : CSS 모듈화

BannerSection, ContentsSection 컴포넌트의 CSS를 모듈화하였습니다.

Related to #40

kelvin-chihyun committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

kelvin-chihyun committed:
🐛 [Fix] : Admin 페이지 버튼 CSS 수정

CSS 모듈화하며, 버튼에 CSS가 잘못 적용되어 수정하였습니다.

Related to #40

ingbinsee commented on src/views/Community.jsx:
```diff
@@ -1,7 +1,67 @@
+import { useNavigate } from 'react-router-dom';
+import { useRecoilValue } from 'recoil';
+import { db } from '../../firebase';
+import BannerSection from '../components/common/BannerSection';
+import NoticeSection from '../components/Community/NoticeSection';
```
P1: 상대경로로 되어 있어 배포 환경에서 파일을 찾을 수 없다고 합니다. 절대경로로 변경 요청드립니다.

kelvin-chihyun replied:
변경했습니다.

ingbinsee commented on src/views/Community.jsx:
```diff
@@ -1,7 +1,67 @@
+import { useNavigate } from 'react-router-dom';
+import { useRecoilValue } from 'recoil';
+import { db } from '../../firebase';
+import BannerSection from '../components/common/BannerSection';
```
P1: 폴더명이 소문자로 되어 있어 찾을 수 없다고 합니다. 대문자로 변경 요청드립니다.

kelvin-chihyun replied:
변경했습니다.

ingbinsee commented on src/components/Admin/Admin.module.css:
P2: 모바일 버전에서 label의 스타일링이 깨집니다. 해당 부분 수정 요청드립니다.
![image](https://github.com/ReOpenSG/ReOpenSG/assets/140426866/2d5056e1-c7b1-474e-856d-5efe043650ee)

P2: 아이디와 비밀번호 form의 너비가 일정하지 않습니다. 해당 부분 수정 요청드립니다.
![image](https://github.com/ReOpenSG/ReOpenSG/assets/140426866/e5e54da4-30c5-4bef-8bcb-00ee8a467da4)

kelvin-chihyun replied:
변경했습니다.

ingbinsee commented on src/components/Community/ButtonSection.jsx:
```diff
@@ -0,0 +1,33 @@
+        {' '}
+        <span className="block font-medium">{isEditing ? '수정하기' : '등록하기'} </span>
+      </button>
+      <button type="button" className={`${styles.button} bg-red-400`} onClick={handleCancel}>
+        {' '}
+        <span className="block font-medium">취소하기</span>
+      </button>
```
P1: 취소하기 버튼 클릭 시 아래와 같은 오류가 발생합니다. 확인 후 수정 요청드립니다.
![image](https://github.com/ReOpenSG/ReOpenSG/assets/140426866/b66fd1f0-ee03-4f75-ad45-9d6c5e047d6e)

kelvin-chihyun replied:
수정하였습니다.

ingbinsee reviewed (changes requested):
아래 파일에 의견 남겨두었습니다. 확인 부탁드립니다.

- Admin.module.css (P2 review 2개)
- ButtonSection.jsx (P1 review)
- Community.jsx (P1 review 2개)

kelvin-chihyun committed:
🌈 [Design] : Admin form css 수정

input, label 간의 간격이 일치하지 않아,
flex-1과 flex-auto 속성을 주어 일치시켰습니다.

kelvin-chihyun committed:
🐛 [Fix] : Date 객체 변경

Firebase에서 제공하는 Timestamp 객체에 toDate 메서드는 prop-types에서 제공되지 않는 type이라 별도로, Date 객체로 계산하여 수정하였습니다.

kelvin-chihyun committed:
📂 [Rename] : 디렉토리명 변경

상대 경로 설정으로 파일 인식에 문제가 있어,
@ alias를 추가하여 절대경로로 인식하게 수정하였습니다.

ryudg reviewed (approved):
확인했습니다.
firebase, recoil 직접 사용해보고 아이디어 생각나면 공유해드리겠습니다.

ingbinsee reviewed (approved):
수정사항 확인했습니다.
